org.lflang/src/org/lflang/generator/PythonGenerator.xtend

```diff
@@ -1699,6 +1698,51 @@ class PythonGenerator extends CGenerator {
         }
     }
     
+    /**
+     * Write a Dockerfile for the current federate as given by filename.
+     * The file will go into src-gen/filename.Dockerfile.
+     * If there is no main reactor, then no Dockerfile will be generated
+     * (it wouldn't be very useful).
+     * @param The root filename (without any extension).
+     */
+    override writeDockerFile(String filename) {
+        if (this.mainDef === null) {
+            return
+        }
+        
+        var srcGenPath = fileConfig.getSrcGenPath
+        val dockerFile = srcGenPath + File.separator + filename + '.Dockerfile'
+        val contents = new StringBuilder()
+        
+        // If a dockerfile exists, remove it.
+        var file = new File(dockerFile)
```

Comment:
If the programmer removes the `main` keyword after a successful build, then the produced docker image will not be removed. We probably want to remove the old image in this case as well (i.e., move 1718:1721 to 1709:1712).